desc/integrals/_bounce_utils.py

```diff
@@ -701,127 +850,323 @@ def interp_to_argmin_hard(h, points, knots, g, dg_dz, method="cubic"):
+    # Our softargmax(x) does the proper shift to compute softargmax(x - max(x)),
+    # but it's still not a good idea to compute over a large length scale, so we
+    # warn in docstring to choose upper sentinel properly.
+    argmin = softargmax(
+        beta * _where_for_fft_argmin(points, ext, g_ext, upper_sentinel),
+        axis=-1,
+    )
+    return jnp.linalg.vecdot(
+        argmin,  # shape is (..., num well, num extrema)
+        interp_rfft2(
+            T.eval1d(ext),
+            ext,
+            h[..., jnp.newaxis, :, :],
+            domain1=(0, 2 * jnp.pi / NFP),
+            axes=(-1, -2),
+        )[..., jnp.newaxis, :],
+        # adding axis to broadcast with num well axis
+    )
 
-    z = jnp.linspace(
-        start=setdefault(start, ppoly.x[0]),
-        stop=setdefault(stop, ppoly.x[-1]),
-        num=num,
+
+# TODO (#568): Generalize this beyond ζ = ϕ or just map to Clebsch with ϕ.
+def get_fieldline(alpha_0, iota, num_transit, period):
+    """Get sequence of poloidal coordinates A = (α₀, α₁, …, αₘ₋₁) of field line.
+
+    Parameters
+    ----------
+    alpha_0 : float
+        Starting field line poloidal label.
+    iota : jnp.ndarray
+        Shape (num rho, ).
+        Rotational transform normalized by 2π.
+    num_transit : float
+        Number of ``period``s to follow field line.
+    period : float
+        Toroidal period after which to update label.
+
+    Returns
+    -------
+    fieldline : jnp.ndarray
+        Shape (num rho, num transit) or broadcastable.
+        Sequence of poloidal coordinates A = (α₀, α₁, …, αₘ₋₁) that specify field line.
+
+    """
+    # Δϕ (∂α/∂ϕ) = Δϕ ι̅ = Δϕ ι/2π = Δϕ data["iota"]
+    return alpha_0 + period * (
+        iota if iota.size == 1 else iota[:, jnp.newaxis]
+    ) * jnp.arange(num_transit)
+
+
+def fourier_chebyshev(theta, iota, alpha, num_transit):
+    """Parameterize θ along field lines ``alpha``.
+
+    Parameters
+    ----------
+    theta : jnp.ndarray
+        Shape (num rho, X, Y) or (X, Y).
+        DESC coordinates θ sourced from the Clebsch coordinates
+        ``FourierChebyshevSeries.nodes(M,N,domain=(0,2*jnp.pi))``.
+        Use the ``Bounce2D.compute_theta`` method to obtain this.
+    iota : jnp.ndarray
+        Shape (num rho, ).
+        Rotational transform normalized by 2π.
+    alpha : float
+        Starting field line poloidal label.
+    num_transit : int
+        Number of toroidal transits to follow field line.
+
+    Returns
+    -------
+    T : PiecewiseChebyshevSeries
+        Set of 1D Chebyshev spectral coefficients of θ along field line.
+        {θ_α : ζ ↦ θ(α, ζ) | α ∈ A} where A = (α₀, α₁, …, αₘ₋₁) is the same
+        field line as ``alpha``. Each Chebyshev series approximates θ over
+        one toroidal transit.
+
+    Notes
+    -----
+    The field line label α changes discontinuously, so the approximation
+    g defined with basis function in (α, ζ) coordinates to some continuous
```

Comment:
This means g doesn't match f if the series is not converged.

Assuming this is theta_DESC. If not, do you have to solve for lambda or fit lambda on alpha, zeta grid.

I am slightly confused as to how quantities are obtained along zeta after you fit a fourier Chebyshev in alpha, zeta
because if the fit is imperfect, the quantities would be discontinuous in zeta as you go to the next field line, right?

Reply:
The only quantity I parameterize in (alpha, zeta) is the DESC poloidal angle itself: theta. Once I know this mapping between coordinate systems, I can directly compute quantities parameterized in theta,zeta along field lines without doing additional root finding.

One reason I chose this strategy rather than a direct representation of quantities in f(alpha, zeta) is because (theta, zeta) representations are more spectrally condensed due to periodicity of the boundary (NFP > 1), whereas (alpha zeta) has no periodicity. Since theta is a coordinate, its fourier spectrum is more independent of the periodicity of the boundary, so unlike some other physical quantity f, it is quite efficient to represent theta directly in field line coordinates. )

> This means g doesn't match f if the series is not converged.

Yes, this is true for any approximation. The point I'm making is this:
-  If you parameterize something in PEST angles, you can compute that quantity along field lines just be evaluating the function along certain $\vartheta$ and $\phi$ values. The quantity you compute using this technique will always be continuous, because   $\vartheta$ and $\phi$ vary continuously as you track the same field line.
- The Clebsch coordinate $\alpha$ does not vary continuously as you track the same field line.  In general, discontinuous changes in $\alpha$ can give discontinuous changes in $f(\alpha, \zeta)$, even if $f$ is a continuous map of (alpha, zeta). We know because of physics that $f$ is continuous along field lines, but for the function approximation in (alpha, zeta) to show this it has to converge sufficiently, as this continuity property is not baked in.  
> because if the fit is imperfect, the quantities would be discontinuous in zeta as you go to the next field line, right?

Yes.
- This isn't something to worry about, as $\vartheta, \phi$ parameterizations suffer from the same issue, its just not as visible (which is what i mention in the docstring).

Reply:
Do you have to do some sort of root finding to obtain theta_DESC on an (alpha, zeta) grid? Because given the colocation points of the alpha, zeta grid theta_PEST is immediately known but theta_DESC would require lambda values at those points, right?

Reply:
Yes and yes